src/commands/general/profile.ts

```diff
@@ -0,0 +1,29 @@
+import responses from "../../constants/responses";
+import type { Command } from "../../types/Command";
+import fetchUser from "../../util/fetchUser";
+import profileEmbed from "../../util/profileEmbed"; 
+export const command: Command = {
+  name: "profile",
+  description: "get the binarysearch profile of a user",
+  options: [
+    {
+    	name: 'query',
+        type: 'STRING',
+        description: 'the username of the user to fetch',
+        required: true,
+    }
+  ],
+  permissions: [],
+  async execute(interaction) {
+    if(interaction.isCommand()) {
```

Comment:
the command handler only passes `CommandInteraction` objects into `commands` so this check is redundant.

Reply:
it is but It already does the check in the command handler [here](https://github.com/binarysus/algo-chan/blob/master/src/internals/commandHandler.ts#L104).

Reply:
oh, my apologies